daemon/response.go

```diff
@@ -248,21 +248,30 @@ func makeErrorResponder(status int) errorResponder {
 
 // A FileStream ServeHTTP method streams the snap
 type FileStream struct {
-	FileName string
+	SnapName string
 	Info     snap.DownloadInfo
 	stream   io.ReadCloser
 }
 
 // ServeHTTP from the Response interface
 func (s FileStream) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
-	w.Header().Set("Content-Type", "application/octet-stream")
+	hdr := w.Header()
+	hdr.Set("Content-Type", "application/octet-stream")
+	hdr.Set("Content-Disposition", s.SnapName)
```

Comment:
that's wrong: it should be `attachment; filename=<the filename>`, assuming the filename is just ascii.

Reply:
Fixed, my bad